src/Blocks/Bookmark.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Notion\Blocks;
+
+use Notion\Common\RichText;
+
+/**
+ * Bookmark block
+ *
+ * @psalm-import-type BlockJson from Block
+ * @psalm-import-type RichTextJson from \Notion\Common\RichText
+ *
+ * @psalm-type BookmarkJson = array{
+ *      bookmark: array{
+ *          url: string,
+ *          caption: RichTextJson[],
+ *      },
+ * }
+ */
+class Bookmark implements BlockInterface
+{
+    private const TYPE = Block::TYPE_BOOKMARK;
+
+    private Block $block;
+
+    private string $url;
+
+    /** @var \Notion\Common\RichText[] */
+    private array $caption;
+
+    /** @param \Notion\Common\RichText[] $caption */
```

Comment:
Instead of using `Type[]` syntax, consider being explicit about the shape of the input.

For example:

 * `RichText[]` means `array<array-key, RichText>`
 * `array<int, RichText>` could be `[0 => new RichText(), 4 => new RichText()]`
 * `list<RichText>` is like `array<int, RichText>`, but prevents gaps in array keys (safer, when considering that a lot of this stuff will be `json_encode()`d